runtime/fastly/builtins/fastly.cpp

```diff
@@ -0,0 +1,461 @@
+// TODO: remove these once the warnings are fixed
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Winvalid-offsetof"
+#pragma clang diagnostic ignored "-Wdeprecated-enum-enum-conversion"
+#include "js/experimental/TypedData.h" // used in "js/Conversions.h"
+#pragma clang diagnostic pop
+#include "../../StarlingMonkey/builtins/web/url.h"
+#include "encode.h"
+#include "fastly.h"
+#include "js/Conversions.h"
+#include "js/JSON.h"
+
+using builtins::web::url::URL;
+using fastly::fastly::Fastly;
+
+namespace {
+
+bool DEBUG_LOGGING_ENABLED = false;
+
+} // namespace
+
+bool debug_logging_enabled() { return DEBUG_LOGGING_ENABLED; }
+
+namespace fastly::fastly {
+
+const JSErrorFormatString *FastlyGetErrorMessage(void *userRef, unsigned errorNumber) {
+  if (errorNumber > 0 && errorNumber < JSErrNum_Limit) {
+    return &fastly_ErrorFormatString[errorNumber];
+  }
+
+  return nullptr;
+}
+
+namespace {
+
+api::Engine *ENGINE;
```

Comment:
Here's another `ENGINE` reference that doesn't originate in the `fetch.h` declaration. Are all the `ENGINE` pointers meant to be to the same value, or should they be different?